src/main/java/io/strimzi/kafka/bridge/config/BridgeConfig.java

```diff
@@ -55,21 +67,40 @@ public HttpConfig getHttpConfig() {
     }
 
     /**
-     * Loads the entire bridge configuration parameters from a related map
+     * Loads the entire bridge configuration parameters from a related map.
+     * Some validation and default values are also applied.
      *
      * @param map map from which loading configuration parameters
      * @return overall bridge configuration
      */
     public static BridgeConfig fromMap(Map<String, Object> map) {
+        validateAndApplyDefaults(map);
         KafkaConfig kafkaConfig = KafkaConfig.fromMap(map);
         HttpConfig httpConfig = HttpConfig.fromMap(map);
-
         return new BridgeConfig(map.entrySet().stream()
                 .filter(e -> e.getKey().startsWith(BridgeConfig.BRIDGE_CONFIG_PREFIX))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)),
                 kafkaConfig, httpConfig);
     }
 
+    private static void validateAndApplyDefaults(Map<String, Object> map) {
+        String metricsTypeValue = (String) map.get(METRICS_TYPE);
```

Comment:
can't you get the enum from the String (as by using `MetricsType.fromString`) method and working with enum values instead of using string "equals" across the method?
You could also use explicit `BridgeConfig.METRICS_TYPE` as done in the `getMetricsType()` method.

Reply:
Yes, I just forgot to update this method. Getting the enum also does validation.